engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
+	jr nz, .exit
+	call TitleScreenSequence
+	ld a, $23
+	ld hl, EffectObjectJumpNoDelay	; obj jump no wait? 8CD13
+	call FarCall_hl
+	call DelayFrame
+	and a
+	ret
+
+.exit
+	scf
+	ret
+
+TitleScreenSequence::
+	ld e, a
+	ld d, $00
+	ld hl, TitleScreenSequenceTable
+	add hl, de
+	add hl, de
+	ldi a, [hl]
+	ld h, [hl]
+	ld l, a
+	jp hl
+	
+TitleScreenSequenceTable::
+dw Start 
+dw LoadPokemonLogo 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw MoveTitle 
+dw MoveTitleEnd 
+dw InitFlashTitle 
+dw FlashTitle
+
+dw PMJapaneseChara 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw PMSubtitle 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber
+
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw Version 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw CopyRight
+
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw HoOh 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber
+
+dw WaitForNextSequence 
+dw PressButtonInit 
+dw TitleScreenInputAndTimeout 
+dw FadeMusicOut
+
+IncreaseSequenceNumber:: ;6025
+	ld hl, wSequence	; wSequence = $CB5E
+	inc [hl]
+	ret
+
+WaitForNextSequence:: ;602A
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence+2
+	ld a, [hl]
+	and a
+	jr z, .next_seq
+	dec [hl]
+	ret
+
+.next_seq
+	call IncreaseSequenceNumber
+	ret
+
+LoadPokemonLogo:: ;603a
+	call PrintPokemonLogo ;6196
+	call IncreaseSequenceNumber ;6025
+	ld a, $01
+	ldh [$FFDE], a
+	ret	
+	
+Start:: ;6045
+	call IncreaseSequenceNumber
+	push de
+	ld de, $002D
+	call PlaySFX	; Play "Swish" sound
+	pop de
+	ld a, $80
+	ld [wSequence+2], a
+	call SetLYOverrides
+	ld a, $43
+	ldh [$FFD0], a
+	ret
+
+MoveTitle:: ;605d
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence+2
+	ld a, [hl]
+	and a
+	jr z, .movetitle_nextseq
+	add $04
+	ld [hl], a
+	ld e, a
+.movetitle_wait
+	ldh a, [rLY]
+	cp $40
+	jr c, .movetitle_wait
+	ld a, e
+	call SetLYOverrides
+	ret
+
+.movetitle_nextseq
+	call IncreaseSequenceNumber
+	ret	
+	
+MoveTitleEnd:: ;607a
+	xor a
+	ldh [$FFD0], a
+	call IncreaseSequenceNumber
+	ld de, $0001
+	call PlayMusic		; Play Title Theme
+	ret
+
+InitFlashTitle:: ;6087
+	call IncreaseSequenceNumber
+	ld a, $1A
+	ld [wSequence+2], a
+	ld a, $06
+	ld [wSequence+3], a
+	ret
+
+FlashTitle:: ;6095
+	ld hl, wSequence+3
+	ld a, [hl]
+	and a
+	jr z, .exit
+	dec [hl]
+	ld a, [wSequence+2]
+	xor $1A
+	ld [wSequence+2], a
+	ldh [rBGP], a
+	call DelayFrame
+	call DelayFrame
+	ret
+
+.exit
+	call IncreaseSequenceNumber
+	ld a, $E4
+	ldh [rBGP], a
+	ret
+
+PMJapaneseChara:: ;60b6
+	call PrintPMJapaneseChara
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+PMSubtitle:: ;60c6
+	call PrintPMSubtitle
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+Version:: ;60d6
```

Comment:
Avoid one-word label names like these. Subfunctions of a bigger "handler" function are usually prefixed with that function (e.g. `TitleScreenMain_Version`). This applies to all other ambiguous or vague labels.